### ISSUE TITLE
[BUG] [regression] `npm start` don't use `server.js` by default

### ISSUE THREAD

**Issue by piranna** [npm/cli#2176]
### Current Behavior:

Since upgrade to npm 7, when executing `npm start` without a `start` script defined in `package.json` I get a `missing script: start` error.

### Expected Behavior:

Running `npm start` without a `start` script defined in `package.json` should exec `./server.js` (more specifically, `node server.js`, due to Windows compatibility), as has been doing since always, and [it's defined in the docs](https://docs.npmjs.com/cli/v6/commands/npm-start):

> If no "start" property is specified on the "scripts" object, it will run `node server.js`.

### Steps To Reproduce:

1. create an empty npm project (create folder, cd into, exec `npm init` with default options)
2. create a `server.js` file with next content:
   ```js
   #!/usr/bin/env node

   console.log('hola')
   ```
3. `chmod +x server.js`
4. `npm start`... and get the error

### Environment:
- OS: Ubuntu 20.10
- Node: 15.1.0
- npm: 7.0.8

**Comment by piranna** [comment]
Content of `package.json`:

```json
{
  "name": "tmp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

**Comment by billyjanitsch** [comment]
This was [fixed in 7.0.9](https://github.com/npm/cli/releases/tag/v7.0.9).

**Comment by piranna** [comment]
Oh, thank you, and sorry for opening the issue, didn't find anything related to this...
